src/client.rs

```diff
@@ -654,6 +654,19 @@ pub trait FilesystemClient: PollClient {
         self.request(request::Metadata { location, path })
     }
 
+    /// Rename a file or directory.
+    ///
+    /// If `to` exists, it must be the same type as `from`.  If `to` is a directory, it must be
```

Comment:
nit: Outside of the littlefs documentation, `type` does not feel to have an obvious meaning.

```suggestion
    /// If `to` exists, it must be the same type (file or directory) as `from`.  If `to` is a directory, it must be
```

Reply:
Fixed.